src/index.js

```diff
@@ -65,8 +66,13 @@ export default function urlJoin(...parts) {
         url += '/';
     }
 
-    // Finally add afterPathname (?queryString)
-    url += afterPathname;
+    // If we have a query object to be handled, lets properly stringify it and append it to the url.
+    // This will override a previous query string in url
+    if (options.query) {
```

Comment:
If the URL already contains a queryString, it will be discarded.

We can solve this by:

- Parse `afterPathname` into an object and merge it with `options.query` and, only then, serialise with `query-string`.

e.g.: `urlJoin('foo', 'bar', '?baz=wow&hello=valdemar', { query: { hello: 'andre' } }` should give `/foo/bar?baz=wow&hello=andre`

- Remove support for query strings as parts and only via `options.query` (this is a breaking change).

The later makes more sense but the former is more powerful in a scenario where we already have a URL with a query string and simply want to add more query parameters.

What do you think?

Reply:
I did the 'override' approach mainly because it was not a breaking change and i was familiar with that approach in [got](https://github.com/sindresorhus/got) module.

Though, i think the first option you gave it is more powerful. I would go with that.